.github/workflows/inference_test.yml

```diff
@@ -0,0 +1,70 @@
+name: Test Store and Inference
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'sllm/serve/**'
+      - 'sllm/cli/**'
+      - 'tests/inference_test/**'
+      - 'entrypoint.sh'
+
+jobs:
+  inference_store_tests:
+    runs-on: self-hosted
+    if: contains(github.event.pull_request.labels.*.name, 'ready-for-gpu-testing')
+    container:
+      image: pytorch/pytorch:2.3.0-cuda12.1-cudnn8-devel
+      options: --gpus all
+    defaults:
+      run:
+          shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          apt-get update
+          apt-get install -y patch git npm wget
+
+      - name: Import Python libraries
+        run: |
+          pip install serverless-llm
+          pip install serverless-llm-store
+          pip install ray vllm
+
+          echo "Creating models directory..."
+          mkdir -p models
+          echo "Models directory created at $(pwd)/models"
+
+          echo "Setting up environment variables..."
+          echo "MODEL_FOLDER=$(pwd)/models" >> $GITHUB_ENV
+          echo "LLM_SERVER_URL=http://127.0.0.1:8343/" >> $GITHUB_ENV
+
+      - name: Setup Environment and Start Services
+        env:
+          MODE: "HEAD"
+        run: |
+          chmod +x entrypoint.sh
+          ./entrypoint.sh
+
+      - name: Test sllm store
```

Comment:
In the previous step, when starting the entrypoint in "HEAD" mode, the `sllm-store-server` is not started, so this test should not have a chance to pass.